node/src/components/storage.rs

```diff
@@ -1,859 +1,578 @@
-        )?;
-
-        Ok(LmdbStorage {
-            block_store: Arc::new(block_store),
-            block_height_store: Arc::new(block_height_store),
-            deploy_store: Arc::new(deploy_store),
-            chainspec_store: Arc::new(chainspec_store),
-        })
-    }
-
-    fn block_store(&self) -> Arc<dyn Store<Value = B>> {
-        Arc::clone(&self.block_store) as Arc<dyn Store<Value = B>>
+impl Default for Config {
+    fn default() -> Self {
+        Config {
+            // No one should be instantiating a config with storage set to default.
+            path: "/dev/null".into(),
+            max_block_store_size: 483_183_820_800,
+            max_deploy_store_size: 322_122_547_200,
+            max_deploy_metadata_store_size: 322_122_547_200,
+        }
     }
+}
 
-    fn block_height_store(&self) -> Arc<dyn BlockHeightStore<B::Id>> {
-        Arc::clone(&self.block_height_store) as Arc<dyn BlockHeightStore<B::Id>>
+impl Config {
+    /// Returns a default `Config` suitable for tests, along with a `TempDir` which must be kept
+    /// alive for the duration of the test since its destructor removes the dir from the filesystem.
+    #[cfg(test)]
+    pub(crate) fn default_for_tests() -> (Self, TempDir) {
+        let tempdir = tempfile::tempdir().expect("should get tempdir");
+        let path = tempdir.path().join("lmdb");
+
+        let config = Config {
+            path,
+            ..Default::default()
+        };
+        (config, tempdir)
     }
+}
 
-    fn deploy_store(&self) -> Arc<dyn DeployStore<Block = B, Deploy = D, Value = D>> {
-        Arc::clone(&self.deploy_store) as Arc<dyn DeployStore<Block = B, Deploy = D, Value = D>>
+impl Display for Event {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Event::StorageRequest(req) => req.fmt(f),
+        }
     }
+}
 
-    fn chainspec_store(&self) -> Arc<dyn ChainspecStore> {
-        Arc::clone(&self.chainspec_store) as Arc<dyn ChainspecStore>
+// Legacy code follows below.
+//
+// The functionality about for requests directly from the incoming network should *not* be present
+// in the validator reactor's routing code. Until it can be cleaned up, this legacy implementation
+// block provides a backwards-compatible interface for this functionality. DO NOT EXPAND, RELY ON OR
+// BUILD UPON THIS CODE.
+
+impl Storage {
+    // Retrieve a deploy from the deploy store to handle a legacy network request.
+    pub fn handle_legacy_direct_deploy_request(&self, deploy_hash: DeployHash) -> Option<Deploy> {
+        // NOTE: This function was formerly called `get_deploy_for_peer` and used to create an event
+        // directly. This caused a dependency of the storage component on networking functionality,
+        // which is highly problematic. For this reason, the code to send a reply has been moved to
+        // the dispatching code (which should be removed anyway) as to not taint the interface.
```

Comment:
> I don't believe this is accurate.
>
> The dependency was on network types: NodeId, which is pervasive across the entire codebase, and NetworkRequest (which isn't actually a network component type). The storage component was sending a response and ignoring the outcome - to my mind that's a dependency on EffectBuilder functionality. The storage component had no expectations or requirements for there to even be a network component functioning.

https://github.com/CasperLabs/casper-node/blob/40a573eb9bc0d49eb58bf1313a72e0f1774a8ae4/node/src/components/storage.rs#L669

Producing `NetworkRequests`, as indicated by the `From` trait bound on the event type is how components are supposed to express dependencies. Requests MUST be answered, otherwise a component might stall. Only announcements are optional.

> How? For testing only, or some other reasons?

Testing is the most evident here, but why would tight coupling not be worse than loose if it comes at little to no cost?

> Regardless of how this discussion pans out, I'm fine for this to not be changed in this PR to allow it to be merged as soon as possible.

We really should give the whole `GetRequest`-system a good, hard look and potentially overhaul it to everyone's satisfaction :)

Reply:
I remain unconvinced, but agree with your final sentence :)